src/ol/interaction/Draw.js

```diff
@@ -333,27 +333,25 @@ class Draw extends PointerInteraction {
            * +-------+------<3>-- (the third point may be anywhere on this line)
            */
 
-          if (coordinates.length > 2) {
+           if (coordinates.length > 2) {
             var first = coordinates[0];
             var second = coordinates[1];
             var third = coordinates[2];
 
             // vector from first to second
             var a_vec = [second[0] - first[0], second[1] - first[1]];
 
-            if (a_vec[1] === 0) {
+            if (a_vec[0] === 0 && a_vec[1] === 0) {
                 // catch the case where the first and second point are equal
-                coordinates = [[first, first, first, first]];
+                coordinates = [[first]];
```

Comment:
This still doesn't have the desired effect. I would like this to "reset" the drawing to the state where only a sinlge point was set and the second is floating beneath the cursor. Instead, the rectangle is created with only `first` as single coordinate but internally the `sketchCoords_` still have 3 points and the drawing is finished on the next click. I had a look how to change this but it took me too long. Can you please try to figure this out?